nessie/src/main/java/org/apache/iceberg/nessie/ParsedTableIdentifier.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.nessie;
+
+import com.dremio.nessie.client.NessieClient;
+import java.time.Instant;
+import java.util.Map;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+public class ParsedTableIdentifier {
+
+  private final TableIdentifier tableIdentifier;
+  private final Instant timestamp;
+  private final String reference;
+
+  /**
+   * container class to hold all options in a Nessie table name.
+   */
+  public ParsedTableIdentifier(TableIdentifier tableIdentifier, Instant timestamp, String reference) {
+    this.tableIdentifier = tableIdentifier;
+    this.timestamp = timestamp;
+    this.reference = reference;
+  }
+
+  public TableIdentifier getTableIdentifier() {
+    return tableIdentifier;
+  }
+
+  public Instant getTimestamp() {
+    return timestamp;
+  }
+
+  public String getReference() {
+    return reference;
+  }
+
+  /**
+   * Convert dataset read/write options to a table and ref/hash.
+   */
+  public static ParsedTableIdentifier getParsedTableIdentifier(TableIdentifier path) {
+    return getParsedTableIdentifier(path, ImmutableMap.of());
+  }
+  /**
+   * Convert dataset read/write options to a table and ref/hash.
+   */
+  public static ParsedTableIdentifier getParsedTableIdentifier(String path, Map<String, String> properties) {
```

Comment:
Does this need properties? The only thing that is used is an optional ref. That could be passed by itself rather than as a map.

Reply:
correct, even further the ref isn't really needed. Thanks for pointing that out, much cleaner now.